src/bytes.rs

```diff
@@ -475,15 +489,15 @@ impl From<Arc<[u8]>> for Bytes<'static> {
         Bytes {
             start: data.as_ref().as_ptr().cast(),
             len: data.len(),
-            data: Arc::into_raw(data) as *const (),
+            owner: Arc::into_raw(data) as *const (),
             capacity,
             vtable: <ptr::NonNull<BytesVt> as From<&'static _>>::from(&ARC_BYTES_VT).cast(),
             marker: core::marker::PhantomData,
         }
     }
 }
 #[cfg(feature = "alloc")]
-impl<'a, T: Sized + AsRef<[u8]> + Send + Sync + 'a> From<Arc<T>> for Bytes<'a> {
+impl<'a, T: Sized + AsRef<[u8]> + Send + Sync + 'static> From<Arc<T>> for Bytes<'static> {
```

Comment:
Since this constructor sets a `release` function in the vtable, `Bytes` is free to assume it fully owns the data (allowing upgrades without copies), even though an `Arc<&'non_static [u8]>` doesn't actually own said data.